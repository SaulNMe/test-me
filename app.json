{
  "expo": {
    "name": "Qualificame",
    "slug": "qualificame-native",
    "privacy": "public",
    "sdkVersion": "33.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splashCustom.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff",

    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    }
  }
}
