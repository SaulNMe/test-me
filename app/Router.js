import React, { Component } from 'react';
import { Platform, Animated, Easing } from 'react-native';
import { createAppContainer, createStackNavigator, createBottomTabNavigator, createSwitchNavigator } from 'react-navigation';
import WelcomeScreen from 'qualificame-native/app/screens/WelcomeScreen';
import LoginScreen from 'qualificame-native/app/screens/LoginScreen';
import SignUpScreen from 'qualificame-native/app/screens/SignUpScreen';
import CreateAccountScreen from 'qualificame-native/app/screens/CreateAccountScreen';
import SelectPlanScreen from 'qualificame-native/app/screens/SelectPlanScreen';
import PaymentScreen from 'qualificame-native/app/screens/PaymentScreen';
import KiosksScreen from 'qualificame-native/app/screens/KiosksScreen';
import CreateKioskScreen from 'qualificame-native/app/screens/CreateKioskScreen';
import KioskDetailScreen from 'qualificame-native/app/screens/KioskDetailScreen';
import KioskConfigurationScreen from 'qualificame-native/app/screens/KioskConfigurationScreen';
import KioskContactDataScreen from 'qualificame-native/app/screens/KioskContactDataScreen';
import KioskOptionsScreen from 'qualificame-native/app/screens/KioskOptionsScreen';
import ReportsScreen from 'qualificame-native/app/screens/ReportsScreen';
import CreateReportScreen from 'qualificame-native/app/screens/CreateReportScreen';
import SummaryReportScreen from 'qualificame-native/app/screens/SummaryReportScreen';
import ReportDetailScreen from 'qualificame-native/app/screens/ReportDetailScreen';
import MailBoxScreen from 'qualificame-native/app/screens/MailBoxScreen';
import MyAccountScreen from 'qualificame-native/app/screens/MyAccountScreen';
import ProPlanScreen from 'qualificame-native/app/screens/ProPlanScreen';
import EnterprisePlanScreen from 'qualificame-native/app/screens/EnterprisePlanScreen';
import ChangePasswordScreen from 'qualificame-native/app/screens/ChangePasswordScreen';
import FaqScreen from 'qualificame-native/app/screens/FaqScreen';
import AlertsScreen from 'qualificame-native/app/screens/AlertsScreen';
import BottomBar from 'qualificame-native/app/components/BottomBar';
import LoadingScreen from 'qualificame-native/app/screens/LoadingScreen';
import SendEmailScreen from 'qualificame-native/app/screens/SendEmailScreen';
import VerificationCodeScreen from 'qualificame-native/app/screens/VerificationCodeScreen';
import ConfirmPasswordScreen from 'qualificame-native/app/screens/ConfirmPasswordScreen';
import AuthLoadingScreen from 'qualificame-native/app/screens/AuthLoadingScreen';
import ScannerScreen from 'qualificame-native/app/screens/ScannerScreen';

import GetStateForAction from 'qualificame-native/app/utils/GetStateForAction';
import TransitionConfiguration from 'qualificame-native/app/utils/TransitionConfiguration';

const RecoverPasswordStack = createStackNavigator({
	SendEmailScreen,
	VerificationCodeScreen,
	ConfirmPasswordScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
})

const AuthModalStackNavigator = createStackNavigator({
	WelcomeScreen,
	LoginScreen,
	RecoverPasswordStack,
	SignUpScreen,
	CreateAccountScreen,
	SelectPlanScreen,
	PaymentScreen,
	LoadingScreen: { 
		screen: LoadingScreen,
		navigationOptions: null
	},
}, {
	transitionConfig: TransitionConfiguration
});

const KioskConfigurationStack = createStackNavigator({
	KioskConfigurationScreen,
	KioskOptionsScreen,
	KioskContactDataScreen
});

const CreateReportStack = createStackNavigator({
	CreateReportScreen,
	SummaryReportScreen,
	LoadingScreen,
}, {
	transitionConfig: TransitionConfiguration,
	navigationOptions: {
		header: null
	}
});

GetStateForAction(CreateReportStack);

const BottomBarNavigator = createBottomTabNavigator({
	KiosksScreen,
	ReportsScreen,
	MailBoxScreen,
	MyAccountScreen
}, {
	initialRouteName: 'KiosksScreen',
	tabBarComponent: BottomBar
});

const CameraModalNavigator = createStackNavigator({
	ScannerScreen
}, {
	mode: 'modal',
	navigationOptions: {
		header: null,
	}
})

const MainStackNavigator = createStackNavigator({
	BottomBarNavigator: {
		screen: BottomBarNavigator,
		navigationOptions: {
			header: null,
		}
	},
	KioskDetailScreen,
	CameraModalNavigator,
	ReportDetailScreen,
	ProPlanScreen,
	EnterprisePlanScreen,
	ChangePasswordScreen,
	FaqScreen
}, {
	initialRouteName: 'BottomBarNavigator',
});

const ModalStack = createStackNavigator({
	MainStackNavigator,
	AlertsScreen,
	CreateKioskScreen,
	KioskConfigurationStack,
	CreateReportStack
}, {
	mode:'modal',
	headerMode: 'none'
});

const MainNavigation = createSwitchNavigator({
	AuthLoading: AuthLoadingScreen,
	Auth: AuthModalStackNavigator,
	App: ModalStack,
}, {
	initialRouteName: 'AuthLoading'
});

export default createAppContainer(MainNavigation);




 




