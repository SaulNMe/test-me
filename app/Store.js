import { createStore, applyMiddleware, compose } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel1 from 'redux-persist/lib/stateReconciler/autoMergeLevel1';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk'
import rootReducer from 'qualificame-native/app/reducers'
import Reactotron from 'qualificame-native/config/ReactotronConfig'

const persistConfig = {
	key: 'root',
	storage: storage,
	stateReconciler: autoMergeLevel1,
	whitelist: ['Auth']
};

const pReducer = persistReducer(persistConfig, rootReducer);

let store = createStore(
	pReducer,
	compose(
		applyMiddleware(thunk),
		Reactotron.createEnhancer()
	)
);

export default store;
export const persistor = persistStore(store);