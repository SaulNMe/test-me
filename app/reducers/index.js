import Events, * as fromEvents from 'qualificame-native/app/reducers/EventsReducer.js';
import Auth, * as fromAuth from 'qualificame-native/app/reducers/AuthReducer.js';
import TokenRefresh, * as fromTokenRefresh from 'qualificame-native/app/reducers/TokenRefreshReducer.js'
import Kiosks, * as fromKiosks from 'qualificame-native/app/reducers/KiosksReducer.js';
import Reports, * as fromReports from 'qualificame-native/app/reducers/ReportsReducer.js';
import Choices, * as fromChoices from 'qualificame-native/app/reducers/ChoicesReducer.js';
import Alerts, * as fromAlerts from 'qualificame-native/app/reducers/AlertsReducer.js';

import { combineReducers } from 'redux'

const rootReducer = combineReducers({
    Auth,
    TokenRefresh,
    Kiosks,
    Reports,
    Events,
    Choices,
    Alerts,
});

export default rootReducer;

//Auth
export const getIsLoggedIn = state =>
	fromAuth.getIsLoggedIn(state.Auth);

export const getIsLoading = state =>
	fromAuth.getIsLoading(state.Auth);

export const getAuthData = state =>
	fromAuth.getAuthData(state.Auth);

export const getUserData = state =>
	fromAuth.getUserData(state.Auth);

//Token Refresh
export const getIsRefreshingToken = state =>
	fromTokenRefresh.getIsRefreshingToken(state.TokenRefresh);

export const getRefreshCall = state =>
	fromTokenRefresh.getRefreshCall(state.TokenRefresh);

//Kiosks
export const getKiosks = state =>
	fromKiosks.getKiosks(state.Kiosks);

export const getKioskById = (state, kioskId) =>
	fromKiosks.getKioskById(state.Kiosks, kioskId)

export const getIsLoadingKiosks = state =>
	fromKiosks.getIsLoadingKiosks(state.Kiosks);

export const getErrorKiosks = state =>
	fromKiosks.getErrorKiosks(state.Kiosks);

//Reports

export const getReports = state =>
	fromReports.getReports(state.Reports);

export const getIsLoadingReports = state =>
	fromReports.getIsLoadingReports(state.Reports);
export const getErrorReports = state =>
	fromReports.getErrorReports(state.Reports);

// Events
export const getEvents = state =>
	fromEvents.getEvents(state.Events);

export const getEventsGroupedByDay = (state, ids) =>
	fromEvents.getEventsGroupedByDay(state.Events, ids);

export const getSelectedMonth = state => 
	fromEvents.getSelectedMonth(state.Events);

export const getEventsSelectedDateRange = state => 
	fromEvents.getEventsSelectedDateRange(state.Events);

// Choices
export const getChoices = state =>
	fromChoices.getChoiceById(state.Choices);

// Alerts 
export const getAlerts = state =>
	fromAlerts.getAlerts(state.Alerts);

export const unReadAlerts = state =>
	fromAlerts.unReadAlerts(state.Alerts);