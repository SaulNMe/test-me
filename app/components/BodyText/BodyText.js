import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './BodyTextStyle';

export default class BodyText extends PureComponent {
	render() {
		const textColor = {color: Colors[this.props.color]};
		const weight = {fontWeight: Fonts.weight[this.props.weight] };
		const align = {textAlign: this.props.align };

		return (
			<View>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</View>
		);
	}
}

	BodyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	BodyText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
