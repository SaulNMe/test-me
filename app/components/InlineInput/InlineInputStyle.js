import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.screen,
	...ApplicationStyles.margins,
	...ApplicationStyles.flexBox,
	flex07: {
		flex: .7
	},
	flex03: {
		flex: .3	
	},
	divider: {
		width: '100%',
		borderBottomWidth: 1,
		borderBottomColor: '#ccc'
	},
	pickerStyle: {
		height: 50,
		width: 100,
		backgroundColor: 'gray',
		flex: 1,
	},
	bgColor: {
		backgroundColor: Colors.white
	},
});
