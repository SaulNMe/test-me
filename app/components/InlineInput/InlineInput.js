import React, { PureComponent } from 'react';
import { 
	Text,
	View,
	TextInput,
} from 'react-native';
import PropTypes from 'prop-types';

import styles from './InlineInputStyle';
import { Colors, Fonts } from 'qualificame-native/app/styles';

import HugeText from 'qualificame-native/app/components/HugeText';
import BodyText from 'qualificame-native/app/components/BodyText';

export default class InlineInput extends PureComponent {

	handleInputChange = (amount) => {
		this.props.onChange(amount, this.props.tag);
	}

	render() {
		return (
			<View style={styles.bgColor}>
				{ this.props.addTopDivider && <View style={[styles.divider, styles.bgColor]}/> }	
				<View style={[styles.row, styles.smallVerticalPadding, styles.alignItemsCenter, styles.baseHorizontalMargin ]}>
					<View style={[styles.flex07, styles.smallRightMargin]}>
						<BodyText 
							text={this.props.text}
							weight="regular"
							color="dark"
						/>
					</View>
					<View style={[styles.flex03, styles.justifyContentFlexEnd, styles.row]}>
						<TextInput
							keyboardAppearance='dark'
							keyboardType="phone-pad"
							onChangeText={this.handleInputChange}
							value={this.props.amount}
							maxLength={3}
							textContentType='telephoneNumber'
							style={
								{
									color:Colors[this.props.color],
									fontSize: Fonts.size.huge,
									fontWeight: Fonts.weight.regular
								}
							}
						/>
						{ this.props.percent && <HugeText
							text= '%'
							color= { this.props.color}
							weight= 'regular'
						/> }
					</View>
				</View>
				{ this.props.addBottomDivider && <View style={styles.divider}/> }	
			</View>
		);
	}
}

	InlineInput.propTypes = {
		text: PropTypes.string,
		amount: PropTypes.string,
		color: PropTypes.string,
		addTopDivider: PropTypes.bool,
		addBottomDivider: PropTypes.bool,
		percent: PropTypes.bool,
	}

	InlineInput.defaultProps = {
		text: "",
		amount: "",
		color: "green",
		addTopDivider: false,
		addBottomDivider: false,
		percent: false
	}
