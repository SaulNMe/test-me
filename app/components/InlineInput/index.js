import InlineInput from './InlineInput';

export default InlineInput;
