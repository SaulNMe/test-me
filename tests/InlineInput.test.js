import React from 'react';
import { shallow } from 'enzyme';

import { Text, TextInput } from 'react-native';
import InlineInput from 'qualificame-native/app/components/InlineInput';
import HugeText from 'qualificame-native/app/components/HugeText';
import BodyText from 'qualificame-native/app/components/BodyText';
import { Colors, Fonts } from 'qualificame-native/app/styles';

function createTestProps (props) {
	return {
		createdAtText: 'created',
		text: 'text test',
		color: 'red',
		//percent: true,
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<InlineInput {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('displays test title as expected', () => {
		expect(wrapper.find(BodyText).prop('text')).toEqual('text test');
	});
	it('displays BodyText dark color', () => {
		expect(wrapper.find(BodyText).prop('color')).toEqual('dark');
	});	
	it('displays BodyText regular weight', () => {
		expect(wrapper.find(BodyText).prop('weight')).toEqual('regular');
	});	
	it('displays the TextInput ', () => {
		expect(wrapper.find(TextInput).exists()).toBeTruthy();
		expect(wrapper.find(TextInput).prop('keyboardType')).toEqual('numeric')
		expect(wrapper.find(TextInput).prop('maxLenght')).toEqual(3)
	});
	it('when HugeText to be falsy', () => {
		expect(wrapper.find(HugeText).exists()).toBeFalsy();
	});
	describe('when HugeText is true', () => {
		beforeEach(() => {
			wrapper = createWrapper({ percent: true });
		});
		it('displays the HugeText', () => {
			expect(wrapper.find(HugeText).exists()).toBeTruthy();
			expect(wrapper.find(HugeText).prop('text')).toEqual('%');
			expect(wrapper.find(HugeText).prop('weight')).toEqual('bold');
			expect(wrapper.find(HugeText).prop('color')).toEqual('green');
		});
	});
});