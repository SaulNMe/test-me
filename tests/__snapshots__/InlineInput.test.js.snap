// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component>
  <Component
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "paddingVertical": 8,
        },
        Object {
          "alignItems": "center",
        },
        Object {
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "flex": 0.7,
          },
          Object {
            "marginRight": 8,
          },
        ]
      }
    >
      <BodyText
        align=""
        color="dark"
        text="text test"
        weight="regular"
      />
    </Component>
    <Component
      style={
        Array [
          Object {
            "flex": 0.3,
          },
          Object {
            "justifyContent": "flex-end",
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        keyboardType="numeric"
        maxLenght={3}
        onChangeText={[Function]}
        style={
          Object {
            "color": "#EB3F3F",
            "fontSize": 32,
            "fontWeight": "700",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </Component>
  </Component>
</Component>
`;
